app/controllers/settings_controller.rb

```diff
@@ -256,6 +256,30 @@ def twitter_disconnect
     return redirect_to "/settings"
   end
 
+  def freenode_callback
+    reader = FreenodeTaxonomyReader.new
+    taxonomy = reader.for_user(params[:user][:freenode_username])
+
+    if taxonomy['LOBSTERS'] == @user.username
+      @user.freenode_username = params[:user][:freenode_username]
+      @user.save!
+      flash[:success] = "Your account has been linked to freenode user #{@user.freenode_username}."
+    elsif taxonomy['LOBSTERS']
+      flash[:error] = "Expected #{@user.username} in the taxonomy, not #{taxonomy['LOBSTERS']}."
```

Comment:
This is not an especially useful error mesage for a user to read. And until I know more about where these taxonomies get filled in from, I'm a little hesitant to drop potential user input into it.

Maybe a more useful error would be something like "Checked on your username, but Freenode told us it's something other than #{@user.username}" because that also handles the case of someone deliberately inputting a different person's username.